### PR TITLE
ci: cleanup resources after e2e tests

### PR DESCRIPTION
**Pull request by apricote**
Remove all deployed resources to avoid running into limit errors.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -102,6 +102,11 @@ jobs:
           source dev/files/env.sh
           make -C test/e2e/kubernetes serial
 
+      - name: Cleanup
+        if: ${{ always() }}
```
Comment (jooola) [inline .github/workflows/test.yml:106]:
```suggestion
        if: always()
```

**Review by jooola — APPROVED** [review]

**Comment by jooola** [comment]
Hmm, does the k8s tests cleanup the volumes if the test fail? Because I am not sure if we clean volumes with the terraform module.

**Comment by apricote** [comment]
We dont, the kubernetes csi tests try to clean those up if possible. I could add a `hcloud volume delete $(hcloud volume list -o noheader -o columns=id)` after taking down the env. (With a check for empty list)

**Comment by jooola** [comment]
> We dont, the kubernetes csi tests try to clean those up if possible. I could add a hcloud volume delete $(hcloud volume list -o noheader -o columns=id) after taking down the env. (With a check for empty list)

Let's see how that goes, if we stumble on a problem again we'll add the command.

**Pull request merged**